sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/AzureSdkCompat/AzureEventSourceLogForwarder.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Diagnostics.Tracing;
+using Azure.Core.Diagnostics;
+using Azure.Core.Shared;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Azure;
+
+namespace Azure.Monitor.OpenTelemetry.AspNetCore.Internals.AzureSdkCompat
+{
+    internal sealed class AzureEventSourceLogForwarder : IHostedService, IDisposable
+    {
+        internal static readonly AzureEventSourceLogForwarder Noop = new AzureEventSourceLogForwarder(null);
+        private readonly ILoggerFactory _loggerFactory;
+
+        private readonly ConcurrentDictionary<string, ILogger> _loggers = new ConcurrentDictionary<string, ILogger>();
+
+        private readonly Func<EventSourceEvent, Exception, string> _formatMessage = FormatMessage;
+
+        private AzureEventSourceListener _listener;
+
+        public AzureEventSourceLogForwarder(ILoggerFactory loggerFactory)
+        {
+            _loggerFactory = loggerFactory;
+        }
+
+        private void LogEvent(EventWrittenEventArgs eventData)
+        {
+            var logger = _loggers.GetOrAdd(eventData.EventSource.Name, name => _loggerFactory!.CreateLogger(ToLoggerName(name)));
+            logger.Log(MapLevel(eventData.Level), new EventId(eventData.EventId, eventData.EventName), new EventSourceEvent(eventData), null, _formatMessage);
+        }
+
+        private static string ToLoggerName(string name)
+        {
+            return name.Replace('-', '.');
+        }
+
+        private static LogLevel MapLevel(EventLevel level)
+        {
+            switch (level)
+            {
+                case EventLevel.Critical:
+                    return LogLevel.Critical;
+                case EventLevel.Error:
+                    return LogLevel.Error;
+                case EventLevel.Informational:
+                    return LogLevel.Information;
+                case EventLevel.Verbose:
+                    return LogLevel.Debug;
+                case EventLevel.Warning:
+                    return LogLevel.Warning;
+                case EventLevel.LogAlways:
+                    return LogLevel.Information;
+                default:
+                    AzureMonitorAspNetCoreEventSource.Log.MapLogLevelFailed(level);
+                    return LogLevel.None;
+            }
+        }
+
+        private static string FormatMessage(EventSourceEvent eventSourceEvent, Exception exception)
+        {
+            return EventSourceEventFormatting.Format(eventSourceEvent.EventData);
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (_loggerFactory != null)
+            {
+                _listener ??= new AzureEventSourceListener((e, s) => LogEvent(e), EventLevel.Verbose);
```

Comment:
Is it possible for this event listener and the one in `sdk/extensions/Microsoft.Extensions.Azure/src/AzureEventSourceLogForwarder.cs` to be active simultaneously?

Reply:
Never mind, next file had an answer.

Reply:
No if users enabled `Microsoft.Extensions.Azure/src/AzureEventSourceLogForwarder` in a [documented way](https://learn.microsoft.com/en-us/dotnet/azure/sdk/logging#map-to-aspnet-core-logging) - there is a check for it and Distro's version of log forwarder will not be added if the original one was enabled.

In theory, users can break it by enabling logs without registering forwarder in the DI, but that's not documented. Also, anything else they do on the DI with Azure SDK clients (without the distro) would result in duplicated logs as well.